language/tools/move-cli/src/main.rs

```diff
@@ -52,10 +55,36 @@ pub enum Command {
             default_value = DEFAULT_SOURCE_DIR,
         )]
         source_files: Vec<String>,
-        /// If set, fail when attempting to typecheck a module that already exists in global storage
-        #[structopt(long = "no-republish")]
-        no_republish: bool,
     },
+    /// Compile and emit Move bytecode for the specified scripts and/or modules
+    #[structopt(name = "compile")]
+    Compile {
+        /// The source files to check
+        #[structopt(
+            name = "PATH_TO_SOURCE_FILE",
+            default_value = DEFAULT_SOURCE_DIR,
+        )]
+        source_files: Vec<String>,
+        /// Emit source map information along with the compiled bytecode
+        #[structopt(long = "save-source-maps")]
```

Comment:
Perhaps worth having a discussion re: whether we want producing source maps to be the default. E.g., I think `move sandbox view <bytecode>` will have nicer output in the default case if we do this.